### ISSUE TITLE
Colorbar levels issue with small range values

### ISSUE THREAD

**Issue by mickaellalande** [lukelbd/proplot#124]
I'm having trouble with colorbar recently, here is an example of it (link with https://github.com/lukelbd/proplot/issues/117? ; I didn't update recently I hope this haven't been solved already, proplot version: 0.4.2):

A first example that works (I purposely multiplied the temperature by 0.01 for having small values for next examples):
```python
import xarray as xr
import proplot as plot

da = xr.tutorial.open_dataset('air_temperature').isel(time=0).air

f, axs = plot.subplots(proj='cyl', axwidth=3)

cmap='ColdHot'
levels=plot.arange(-10,10, 1)

m = axs.pcolormesh(((da-273.15)*0.01), cmap=cmap, levels=levels, colorbar='r')

axs.format(labels=True, coast=True, borders=True,
    latlim=(da.lat.min(), da.lat.max()), lonlim=(da.lon.min(), da.lon.max())
)
```
![temp_1](https://user-images.githubusercontent.com/20254164/75169595-1bce4500-5729-11ea-801f-672babbbadd1.png)

Then with
```python
levels=plot.arange(-2, 2, 0.1)
```
![temp_2](https://user-images.githubusercontent.com/20254164/75169648-34d6f600-5729-11ea-8820-ad0b5ca60734.png)

It still works but already the 0 is not good... 

And if I go for a smaller range:
```python
levels=plot.arange(-1, 1, 0.1)
```
this time I have an error:

```python
[...]
~/miniconda3/lib/python3.7/site-packages/proplot/axistools.py in __call__(self, x, pos)
    468                 # Hard limit of MAX_DIGITS sigfigs
    469                 string = ('{:.%df}' % min(
--> 470                     abs(np.log10(x) // 1), MAX_DIGITS)).format(x)
    471                 continue
    472             break

ValueError: cannot convert float NaN to integer
```

Knowing that it is the levels I need to use for another application, so I'm stuck for now.

**Comment by lukelbd** [comment]
This can be temporarily fixed by applying a [`SimpleFormatter`](https://proplot.readthedocs.io/en/latest/api/proplot.axistools.SimpleFormatter.html#proplot.axistools.SimpleFormatter) with explicit decimal precision (see #117), e.g. as follows:

```python
ax.colorbar(..., formatter=('simple', 3))
```

Note that by default, this formatter still trims trailing zeros (this can also be turned off with the `zerotrim` argument).

I'm pretty busy this week but will try to fix this by next weekend; this issue has high priority.

**Comment by mickaellalande** [comment]
> ax.colorbar(..., formatter=('simple', 3))

Ok, great thanks for the trick! It works fine so far like this :smile:

**Comment by lukelbd** [comment]
This issue should be fixed for >=99% of cases by 9b7f89fd, specifically the lines:

https://github.com/lukelbd/proplot/blob/9b7f89fdad3a8e127570aeccbe138c9c089d1fff/proplot/ticker.py#L467-L491
